specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL/preview/2020-02-14-privatepreview/examples/ServerGetWithVnet.json

```diff
@@ -0,0 +1,38 @@
+{
+  "parameters": {
+    "serverName": "pgtestsvc1",
+    "resourceGroupName": "testrg",
+    "api-version": "2020-02-14-privatepreview",
+    "subscriptionId": "ffffffff-ffff-ffff-ffff-ffffffffffff"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "properties": {
+          "administratorLogin": "cloudsa",
+          "version": "12",
+          "userVisibleState": "Ready",
+          "userVisibleHAState": "NotEnabled",
+          "fullyQualifiedDomainName": "pgtestsvc4.postgres.database.azure.com",
+          "displayName": "demosingleserver",
+          "publicIpAddressRequested": true,
+          "serverEdition": "GeneralPurpose",
+          "vCores": 4,
+          "storageQuotaInMb": 524288,
+          "vnetInjArgs": {
+            "delegatedSubnetName": "test-subnet",
+            "delegatedVnetID": "6ed504a3-24b8-4ba8-ac1f-7a1780195437",
+            "delegatedVnetName": "test-vnet"
+          }
```

Comment:
The recommended pattern for things like this is to either return a string with the fully-qualified subnet resource ID or an object with `id` property inside that is the subnet resource ID. THat way the user can do a GET and get all the properties of the V-net and not just a couple that you chose.

Reply:
This is due to the DNC service takes the resource guid of the vnet. In backend we don’t call ARM to get the resources, so there’s no fully-qualified ARM resource ID. We just persist this resource guid and send to DNC to register the vnet.

Here’s some documentation for DNC APIs: https://microsoft.sharepoint.com/teams/WAG/AzureNetworking/SitePages/DNC%20-%20Delegated%20Network%20Controller.aspx. (=> 4. Deposit the Vnet)

Reply:
I see. Makes sense.

---
In reply to: [451863323](https://github.com/Azure/azure-rest-api-specs/pull/10020#discussion_r451863323) [](ancestors = 451863323)